pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -283,6 +283,11 @@ func (c *Cluster) Start() error {
 		return errors.Wrapf(err, "failed to reconcile key rotation cron jobs")
 	}
 
+	err = c.postReconcileUpdateOSDProperties(updateConfig.osdDesiredState)
```

Comment:
For my understanding: 

This waits for OSDs to be updated/upgraded, and then changes the device class after, right?

Why do it this way instead of changing the class before updating the OSD deployment? I'm guessing because it would impact the ability to update/upgrade in a timely fashion because data movement would start after the update?

Is there a risk that this change will cause running OSDs to enter error state? It seems to me like probably not, but worth checking.

Reply:
The only downside of this approach I see is that if any external tools are looking at the OSD labels, they will be different from the actual applied settings in Ceph, until applied at the end of the reconcile. This seems like normal/acceptable behavior to have some disconnect of desired and actual state while the reconcile is still in progress. 

I proposed this approach as an optimization, instead of updating the device class during individual OSD updates. The `ceph osd df` call that retrieves the needed device class metadata pulls it for all OSDs, which would be very inefficient to call again and again for individual OSDs, especially when it is rarely going to detect that a change is needed.

Reply:
I see. The deployment change is just a label then?

As long as changing the device classes doesn't risk the OSD running in the deployment to enter a failed state, it seems to me like it doesn't matter where the change happens.

Reply:
Yes, it's just a label change on the deployment/pod spec. It does cause a pod restart which isn't necessary from ceph's perspective.